src/main/java/bob/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package bob;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> tasklist;
+
+    public TaskList() {
+        this.tasklist = new ArrayList<>();
+    }
+
+    public ToDo todo(String name) {
+        ToDo todo = new ToDo(name);
+        this.tasklist.add(todo);
+        return todo;
+    }
+
+    public Deadline deadline(String name, String by) {
+        Deadline deadline = new Deadline(name, by);
+        this.tasklist.add(deadline);
+        return deadline;
+    }
+
+    public Event event(String name, String from, String to) {
+        Event event = new Event(name, from, to);
+        this.tasklist.add(event);
+        return event;
+    }
+
+    public void mark(int idx) {
+        if (idx < 1 || idx > this.tasklist.size()) {
+            throw new TaskIndexException(String.valueOf(idx));
+        }
+
+        this.tasklist.get(idx - 1).complete();
+    }
+
+    public void unmark(int idx) {
+        if (idx < 1 || idx > this.tasklist.size()) {
+            throw new TaskIndexException(String.valueOf(idx));
+        }
+
+        this.tasklist.get(idx - 1).uncomplete();
+    }
+
+    public Task delete(int idx) {
+        if (idx < 1 || idx > this.tasklist.size()) {
+            throw new TaskIndexException(String.valueOf(idx));
+        }
+
+        Task deleted = this.tasklist.get(idx - 1);
+        this.tasklist.remove(idx - 1);
+        return deleted;
+    }
```

Comment:
It would be clearer if you could specify exactly what the methods are doing. For example, `mark()` could be `markTask()` etc.